File: src/com/alxminyaev/neural_network/Layer.java
package com.alxminyaev.neural_network;

import java.util.ArrayList;

abstract class Layer {
    private int numberOfNeuronsInLayer;
    private ArrayList<Neuron> listOfNeural;

    public int getNumberOfNeuronsInLayer() {
        return numberOfNeuronsInLayer;
    }

    public void setNumberOfNeuronsInLayer(int numberOfNeuronsInLayer) {
        this.numberOfNeuronsInLayer = numberOfNeuronsInLayer;
    }

    public ArrayList<Neuron> getListOfNeural() {
        return listOfNeural;
    }

    public void setListOfNeural(ArrayList<Neuron> listOfNeural) {
        this.listOfNeural = listOfNeural;
    }

    abstract void initLayer();

    abstract void printLayer();
}

File: src/com/alxminyaev/neuralnetwork/Layer.kt
package com.alxminyaev.neuralnetwork

import java.util.ArrayList

abstract class Layer(var numberOfNeuronsInLayer: Int) {

    private var listOfNeural: ArrayList<Neuron> = ArrayList()

    init {
        for (i in 0 until this.numberOfNeuronsInLayer) {
            listOfNeural.add(Neuron())
        }
    }

    fun getListOfNeural(): ArrayList<Neuron> {
        return listOfNeural
    }

    abstract fun initLayer()

    abstract fun printLayer()
}